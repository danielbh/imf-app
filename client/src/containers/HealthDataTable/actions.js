import uuid from 'node-uuid';

import {
  ADD_ENTRY,
  DELETE_ENTRY
} from './constants';

export const addEntry = ({ date, duration, weight, bodyFat }) => ({
  type: ADD_ENTRY,
  date: new Date(date).getTime(),
  duration,
  weight,
  bodyFat,
  id: uuid.v4() // TODO: Change to server generated id later
});

export const deleteEntries = ids => ({
  type: DELETE_ENTRY,
  ids
});
