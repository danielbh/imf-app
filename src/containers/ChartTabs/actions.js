import {
  SELECT_DURATION,
  SELECT_WEIGHT,
  SELECT_BODY_FAT,
} from './constants'

export const selectTab = (key) => {
  switch(key) {
    case 2:
      return {type: SELECT_WEIGHT };
    case 3:
      return { type: SELECT_BODY_FAT };
    default:
      return { type: SELECT_DURATION };
  }
};