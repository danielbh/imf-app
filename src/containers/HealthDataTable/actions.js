import {
  ADD_ENTRY,
  DELETE_ENTRY,
} from './constants';

import uuid from 'node-uuid'

export const addEntry = ({ date, duration, weight, bodyFat }) => ({
  type: ADD_ENTRY,
  date,
  duration,
  weight,
  bodyFat,
  id: uuid.v4() // TODO: Change to server generated id later
});

export const deleteEntries = (ids) => ({
  type: DELETE_ENTRY,
  ids
});
