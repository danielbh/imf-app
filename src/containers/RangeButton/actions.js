/**
 * Created by danielhollcraft on 9/8/17.
 */
import { SET_DATE_RANGE } from '../App/constants'

export const setDateRange = (range = 'ALL') => ({
  type: SET_DATE_RANGE,
  range
});

