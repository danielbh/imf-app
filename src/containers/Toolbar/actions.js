/**
 * Created by danielhollcraft on 9/8/17.
 */

import {
  SET_WEEK_RANGE,
  SET_MONTH_RANGE,
  SET_THREE_MONTH_RANGE,
  SET_YEAR_RANGE,
  SET_ALL_RANGE
} from './constants'

export const selectWeek = () => ({
  type: SET_WEEK_RANGE
});

export const selectMonth = () => ({
  type: SET_MONTH_RANGE
});

export const selectThreeMonth = () => ({
  type: SET_THREE_MONTH_RANGE
});

export const selectYear = () => ({
  type: SET_YEAR_RANGE
});

export const selectAll = () => ({
  type: SET_ALL_RANGE
});