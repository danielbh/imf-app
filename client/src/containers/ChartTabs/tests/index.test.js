import React from 'react';
import { shallow } from 'enzyme';
import { ChartTabs, mapDispatchToProps } from '../index'
import { selectTab } from "../actions";

const ChartTabsComponent =
  handleSelect => shallow(<ChartTabs handleSelect={handleSelect} />);

describe('<ChartTabs />', () => {
  it('matches snapshot', () => {
    expect(shallow(<ChartTabs />)).toMatchSnapshot();
  });

  it('passes down handleSelect eventHandler', () => {
    const handleSelect = jest.fn();
    const wrapper = ChartTabsComponent(handleSelect);
    wrapper.props().onSelect(1);
    expect(handleSelect).toHaveBeenCalledWith(1)
  });


  describe('mapDispatchToProps', () => {
    describe('handleSelect', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.handleSelect).toBeDefined();
      });

      it('should dispatch handleSelect when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.handleSelect(1);
        expect(dispatch).toHaveBeenCalledWith(selectTab(1));
      });
    });
  });
});
