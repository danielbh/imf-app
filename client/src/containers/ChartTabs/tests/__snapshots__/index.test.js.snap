// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartTabs /> matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Uncontrolled(Tabs)
    defaultActiveKey={1}
    id="tab-id-required-for-accessibility"
    onSelect={undefined}
>
    <Tab
        eventKey={1}
        title="Duration"
    />
    <Tab
        eventKey={2}
        title="Weight (kg)"
    />
    <Tab
        eventKey={3}
        title="Body Fat %"
    />
</Uncontrolled(Tabs)>,
  "nodes": Array [
    <Uncontrolled(Tabs)
      defaultActiveKey={1}
      id="tab-id-required-for-accessibility"
      onSelect={undefined}
>
      <Tab
            eventKey={1}
            title="Duration"
      />
      <Tab
            eventKey={2}
            title="Weight (kg)"
      />
      <Tab
            eventKey={3}
            title="Body Fat %"
      />
</Uncontrolled(Tabs)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ChartTabs />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Uncontrolled(Tabs)
          defaultActiveKey={1}
          id="tab-id-required-for-accessibility"
          onSelect={undefined}
>
          <Tab
                    eventKey={1}
                    title="Duration"
          />
          <Tab
                    eventKey={2}
                    title="Weight (kg)"
          />
          <Tab
                    eventKey={3}
                    title="Body Fat %"
          />
</Uncontrolled(Tabs)>,
        "_debugID": 2,
        "_renderedOutput": <Uncontrolled(Tabs)
          defaultActiveKey={1}
          id="tab-id-required-for-accessibility"
          onSelect={undefined}
>
          <Tab
                    eventKey={1}
                    title="Duration"
          />
          <Tab
                    eventKey={2}
                    title="Weight (kg)"
          />
          <Tab
                    eventKey={3}
                    title="Body Fat %"
          />
</Uncontrolled(Tabs)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ChartTabs />,
}
`;
